### ISSUE TITLE
Redesign the error format in JSON response

### ISSUE THREAD

**Issue by tkdchen** [Nitrate/Nitrate#504]
Lots of code returning a JSON response contains error messages if something wrong during the process. The problem is the format of those error messages is not clear and consistent. This issue is for redesign the format and refactor the relative both Python and JavaScript code.

Start after #483

**Comment by tkdchen** [comment]
In the PRs related to #483, the JSON response will have different content from different views. However, basically there is a uniform format that has one attribute `message` which could be a string or list of strings. And generally, if everything did well in a view and nothing needs to be returned, the response body is an emtpy `{}`. This works well for now, so close this issue. Reopen or file a new issue if the error format has to be in another different format.
